### PR TITLE
feat: handling and initial commands

### PR DESCRIPTION
**Pull request by almostSouji**
- Repo setup
- Initial handlers
- Move discord.js specific commands from Cerberus to Toolkit

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"root": true,
+	"extends": ["neon/common", "neon/node", "neon/typescript", "neon/prettier"],
+	"parserOptions": {
+		"project": "./tsconfig.eslint.json"
+	},
+	"rules": {
+		"@typescript-eslint/no-unused-vars": 0,
```
Comment (iCrawl) [inline .eslintrc.json]: Neon has this already disabled.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# secrets
+.env
+
+# build flow
+dist/
+node_modules
```
Comment (iCrawl) [inline .gitignore]:
```suggestion
# Packages
node_modules/

# Log files
logs/
*.log
npm-debug.log*

# Runtime data
pids
*.pid
*.seed

# Env
.env

# Dist
dist/

# Miscellaneous
.tmp/
.vscode/*
!.vscode/extensions.json
!.vscode/settings.json
.idea/
.DS_Store
.turbo
tsconfig.tsbuildinfo
coverage/

# yarn
.pnp.*
.yarn/*
!.yarn/patches
!.yarn/plugins
!.yarn/releases
!.yarn/sdks
!.yarn/versions

# Cache
.prettiercache
.eslintcache

```

Has maybe a few obsolete things, but covers more cases.

File: LICENSE [hunk 3]
```diff
@@ -0,0 +1,190 @@
+    Apache License
```
Comment (iCrawl) [inline LICENSE]: What happened here

File: package.json [hunk 4]
```diff
@@ -0,0 +1,59 @@
+{
+	"name": "discord.js-support-toolkit",
+	"version": "0.0.0",
```
Comment (iCrawl) [inline package.json]:
```suggestion
	"version": "0.1.0",
```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+	"description": "Utility bot for the discord.js discord server",
+	"scripts": {
+		"build": "rimraf dist && tsc",
```
Comment (iCrawl) [inline package.json]: Check how I use `swc` here: https://github.com/Naval-Base/yuudachi/blob/main/packages/yuudachi/package.json#L6-L10

https://github.com/Naval-Base/yuudachi/blob/main/packages/yuudachi/.swcrc

File: package.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+		"start": "env-cmd node --enable-source-maps --es-module-specifier-resolution=node dist/index.js",
+		"deploy": "node --unhandled-rejections=strict dist/index.js",
+		"lint": "eslint src --ext .ts",
+		"lint:fix": "eslint src --ext .ts --fix",
```
Comment (iCrawl) [inline package.json]:
```suggestion
		"lint": "prettier --check . && eslint src --ext ts",
		"format": "prettier --write . && eslint src --ext ts --fix",
		"fmt": "yarn format",
```

File: package.json [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+		"discordapp",
+		"discord.js",
+		"slashcommand",
+		"utilities"
+	],
+	"dependencies": {
+		"dayjs": "^1.11.5",
+		"discord.js": "^14.5.0",
+		"dotenv": "^16.0.3",
+		"env-cmd": "^10.1.0",
```
Comment (iCrawl) [inline package.json]: Both?

File: package.json [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+		"readdirp": "^3.6.0",
+		"reflect-metadata": "^0.1.13",
+		"tsyringe": "^4.7.0",
+		"undici": "^5.11.0"
+	},
+	"devDependencies": {
+		"@commitlint/cli": "^17.1.2",
+		"@commitlint/config-angular": "^17.1.0",
+		"@types/node": "^18.8.3",
+		"@types/node-fetch": "^2.6.2",
```
Comment (iCrawl) [inline package.json]:
```suggestion
```

Not needed.

File: package.json [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+		"@types/pino": "^7.0.5",
+		"@typescript-eslint/eslint-plugin": "^5.39.0",
+		"@typescript-eslint/parser": "^5.39.0",
```
Comment (iCrawl) [inline package.json]:
```suggestion
```

Not needed for neon.

File: package.json [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+		"eslint": "^8.24.0",
+		"eslint-config-neon": "^0.1.34",
+		"eslint-config-prettier": "^8.5.0",
+		"eslint-plugin-tsdoc": "^0.2.17",
```
Comment (iCrawl) [inline package.json]:
```suggestion
```

Not needed for neon.

File: package.json [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+		"husky": "^8.0.1",
+		"lint-staged": "^13.0.3",
+		"prettier": "^2.7.1",
+		"prettier-eslint": "^15.0.1",
```
Comment (iCrawl) [inline package.json]:
```suggestion
```

File: src/types/UrlScanTypes.ts [hunk 12]
```diff
@@ -0,0 +1,36 @@
+export type ScanData = {
```
Comment (JPBM135) [inline src/types/UrlScanTypes.ts]: Where this types are used?

File: src/commands/userinfo.ts [hunk 13]
```diff
@@ -0,0 +1,290 @@
...
+	italic,
+	type User,
+	time,
+	TimestampStyles,
+	ApplicationFlagsBitField,
+} from 'discord.js';
+import kleur from 'kleur';
+import { type InteractionParam, type CommandMethod, type ArgsParam, Command } from '../Command.js';
+import type { UserinfoContextCommand } from '../interactions/context/userinfoContext.js';
+import type { UserinfoCommand } from '../interactions/slash/userinfo.js';
```
Comment (iCrawl) [inline src/commands/userinfo.ts]: Shouldn't this be `UserInfo`

File: src/types/DataTypes.ts [hunk 14]
```diff
@@ -0,0 +1,13 @@
+export type GuildMetrics = {
```
Comment (JPBM135) [inline src/types/DataTypes.ts]: Same as above

File: src/deploy.ts [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+import {
+	SnowflakeInfoCommand,
+	BitfieldLookupCommand,
+	IntentsLookupContextCommand,
+	UserinfoCommand,
+	UserinfoContextCommand,
+} from './interactions/index.js';
+import { logger } from './util/logger.js';
+
+const rest = new REST({ version: '9' }).setToken(process.env.DISCORD_TOKEN!);
```
Comment (iCrawl) [inline src/deploy.ts]:
```suggestion
const rest = new REST({ version: '10' }).setToken(process.env.DISCORD_TOKEN!);
```

**Review by JPBM135 — CHANGES_REQUESTED** [review]

File: src/events/threadCreate.ts [hunk 16]
```diff
@@ -0,0 +1,32 @@
+import { setTimeout as wait } from 'node:timers/promises';
+import { Events, Client } from 'discord.js';
+import { injectable } from 'tsyringe';
+import type { Event } from '../Event.js';
+
+const ASSISTCHANNELS = ['986520997006032896', '998942774994927646'];
```
Comment (iCrawl) [inline src/events/threadCreate.ts]: Export to constants.

File: src/interactions/ArgumentsOf.ts [hunk 17]
```diff
@@ -0,0 +1,103 @@
+/* import type { APIGuildMember, APIPartialChannel, APIRole, Permissions, APIAttachment } from 'discord.js'; */
```
Comment (iCrawl) [inline src/interactions/ArgumentsOf.ts]: Make sure this file is up 2 date with yuudachi.

File: src/interactions/InteractionOptions.ts [hunk 18]
```diff
@@ -0,0 +1,67 @@
+import {
```
Comment (iCrawl) [inline src/interactions/InteractionOptions.ts:1]: Same here.

**Review by iCrawl — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (JPBM135) [inline package.json:41]: This types are unused

**Review by JPBM135 — CHANGES_REQUESTED** [review]

File: src/util/index.ts [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+	return resText.length === text.length ? resText : `${resText.trim()}...`;
+}
+
+export const LIMIT_EMBED_DESCRIPTION = 4_048 as const;
+export const LIMIT_EMBED_TITLE = 256 as const;
+export const LIMIT_EMBED_FIELDS = 25 as const;
+export const LIMIT_EMBED_FIELD_NAME = 256 as const;
+export const LIMIT_EMBED_FIELD_VALUE = 1_024 as const;
+export const LIMIT_EMBED_AUTHOR_NAME = 256 as const;
+export const LIMIT_EMBED_FOOTER_TEXT = 2_048 as const;
```
Comment (JPBM135) [inline src/util/index.ts]: This shouldn't be on consts too?

Comment (almostSouji) [reply src/util/index.ts]: it's contextual where it's used, which makes sense imo

File: .gitattributes [hunk 20]
```diff
@@ -0,0 +1 @@
+* text=auto eol=lf
```
Comment (iCrawl) [inline .gitattributes]: Newline at the end

File: .gitignore [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+
+# Cache
+.prettiercache
+.eslintcache
+
```
Comment (iCrawl) [inline .gitignore]: Can prolly remove this explicit newline but make sure theres one for EOF

File: src/util/constants.ts [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+	DiscordSuccess = 0x3ba55d,
+	DiscordDanger = 0xed4245,
+}
+
+export const DATE_FORMAT_LOGFILE = "YYYY-MM-DD_HH-mm-ss";
+export const DATE_FORMAT_WITH_SECONDS = "YYYY/MM/DD HH:mm:ss";
+
+export const DISCORD_USER_FLAG_SPAMMER = 1 << 20;
+
+export const TAB = " \u200B \u200B \u200B " as const;
```
Comment (iCrawl) [inline src/util/constants.ts]: Are the spaces start and end here necessary?

File: src/util/index.ts [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
```
Comment (iCrawl) [inline src/util/index.ts]: Maybe hoist them at the top at least.

File: src/util/index.ts [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+
+/**
+ * Truncate the provided EMBED
```
Comment (iCrawl) [inline src/util/index.ts]: Any reason those instances of embed are capitalized?

**Review by iCrawl — CHANGES_REQUESTED** [review]

File: src/util/constants.ts (hunk 22 above)
Comment (almostSouji) [reply src/util/constants.ts]: true, those are trimmed by discord

**Pull request merged**